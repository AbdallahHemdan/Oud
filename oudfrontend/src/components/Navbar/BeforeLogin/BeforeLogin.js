import React from 'react';
import { Link } from 'react-router-dom'

/**
 * Function to render rhe right part of the navbar if the use in guest mode (not logged in)
 * 
 * @function
 * 
 * @param {object} props 
 * 
 * @returns {JSX}
 */
function BeforeLogin(props) {
    return (
        <form
            className="form-inline my-2 my-lg-0"
            data-testid="before-login"
        >
            <Link
                to="/signin"
                className="signup-signin-link"
                data-testid="login-link"
            >
                <button
                    className="btn oud-btn my-2 my-sm-0 mr-3 login-signup-btn signin"
                    type="submit"
                    data-testid="login-btn"
                >
                    Sign in
                </button>
            </Link>
            <Link
                to="/signup"
                className="signup-signin-link"
                data-testid="signup-link"
            >
                <button
                    className="btn oud-btn my-2 my-sm-0 mr-3 login-signup-btn signup"
                    type="submit"
                    data-testid="signup-btn"
                >
                    Sign up
                </button>
            </Link>
        </form >
    )
}

export { BeforeLogin }