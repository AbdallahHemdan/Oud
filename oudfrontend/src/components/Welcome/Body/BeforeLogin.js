import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import Back from './../../../assets/images/2685063.jpg';
import oud from './../../../assets/images/oud.png';

class BeforeLogin extends Component {
  render() {
    return (
      <div className="BodyStyle">
        <img
          src={Back}
          className="backGround"
          alt="background"
          data-testid="background"
        ></img>
        <div className="welcomeTextGeust">
          <img src={oud} alt="oud" className="oudLogoBody" />
          <h1 data-testid="firstText">Music for everyone.</h1>
          <h6 data-testid="firstText">
            Millions of songs. No credit card needed.
          </h6>
          <Link to="/SignUP">
            <button className="getOudBtn" data-testid="getOudBtn">
              SIGNUP NOW
            </button>
          </Link>
        </div>
      </div>
    );
  }
}

export default BeforeLogin;
