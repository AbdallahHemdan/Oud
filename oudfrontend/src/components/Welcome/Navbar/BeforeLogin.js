import React, { Component } from 'react'
import { Link } from 'react-router-dom';

class BeforeLogin extends Component {
  render() {
    return (
      <React.Fragment>
        <li className="nav-item">
          <Link data-testid="SignUP" to="/signup" className="nav-link">
            Sign up
          </Link>
        </li>
        <li className="nav-item">
          <Link
            data-testid="Login"
            className="nav-link"
            to="/signin"
          >
            Log in
          </Link>
        </li>
      </React.Fragment>
    )
  }
}

export default BeforeLogin
