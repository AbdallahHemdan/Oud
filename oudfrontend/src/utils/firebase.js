import firebase from "firebase";

const firebasConfig = {
  apiKey: "AIzaSyBZVUambAFEB-ywzAZTAPP-tSkbYXSEWc4",
  authDomain: "oud-notifications.firebaseapp.com",
  databaseURL: "https://oud-notifications.firebaseio.com",
  projectId: "oud-notifications",
  storageBucket: "oud-notifications.appspot.com",
  messagingSenderId: "681943816425",
  appId: "1:681943816425:web:9149a8834b2ff0f4623878"
};
// Initialize Firebase
firebase.initializeApp(firebasConfig);

export default firebase;
