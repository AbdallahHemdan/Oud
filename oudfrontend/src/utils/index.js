import axios from 'axios'
import { base } from '../config/environment'
const findByTestAttr = (component, val) => {
    return component.find(`[data-testid="${val}"]`);
}

function addToLikedSongs(id) {
    axios.put(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);

        })
        .catch(function (error) {
            console.log(error);
        });
}
function removeLikedSong(id) {
    axios.delete(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);
        })
        .catch(function (error) {
            console.log(error);
        });
}
export { findByTestAttr, addToLikedSongs, removeLikedSong }
