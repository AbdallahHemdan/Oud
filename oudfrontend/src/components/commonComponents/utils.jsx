import axios from "axios";
import { base } from "../../config/environment"
import {config} from "../../utils/auth"
/**
 * @file this file contains three functions used in Playlist, Album, LikedSongs 
 **/
export function addSong(playlistId, track){
  axios
  .post(`${base}/playlists/${playlistId}/${track}`, config)
  .then((response) => {
  })
  .catch((error) => {
  });
}