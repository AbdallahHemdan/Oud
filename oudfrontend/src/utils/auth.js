import jwtDecode from "jwt-decode";
import { getRequest } from "./requester";
import { base } from "./../config/environment";

function getToken() {
  return localStorage.getItem("accessToken");
}

const config = {
  headers: { authorization: `Bearer ${getToken()}` }
};

function Auth() {
  try {
    if (!getToken()) return null;
    let decoded = jwtDecode(getToken());
    console.log(decoded.id);
    return decoded.id;
  } catch {
    return null;
  }
}
function isLoggedIn() {
  return getToken() ? true : false;
}

function isArtist() {
  return getRequest(`${base}/me`)
    .then(response => {
      return response.data.role === "artist";
    })
    .catch(error => {
      console.log(error);
    });
}
const _config = config;
export { _config as config };
export const token = getToken();
const _Auth = Auth;
export { _Auth as Auth };
const _isLoggedIn = isLoggedIn;
export { _isLoggedIn as isLoggedIn };
const _isArtist = isArtist;
export { _isArtist as isArtist };
