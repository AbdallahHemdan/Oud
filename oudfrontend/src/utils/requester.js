import axios from "axios";
import { isLoggedIn } from "./../utils/auth";
const config = {
  headers: {
    authorization: `Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjVlOTA3ZGIwYTA2NDVmNDU4MTYwNzYwNiIsImlhdCI6MTU5MTYyMTQxOSwiZXhwIjoxNTk0MjEzNDE5fQ.fj3N3Pc89Pf_xlt7fGmXw1SINTecUB4-y3pihAAPjC8`,
    "Access-Control-Allow-Origin": "*"
  }
};
const getRequest = endpoint => {
  return axios.get(endpoint, config);
};
/**
 * Axios DELETE request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @returns {object}
 */
const deleteRequest = endpoint => {
  return axios.delete(endpoint, config);
};
/**
 * Axios PUT request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const putRequest = (endpoint, body = {}) => {
  return axios.put(endpoint, body, config);
};
/**
 * Axios POST request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const postRequest = (endpoint, body = {}) => {
  return axios.post(endpoint, body, config);
};
/**
 * Axios PATCH request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const patchRequest = (endpoint, body = {}) => {
  return axios.patch(endpoint, body, config);
};

// const isArtist = () => {
//   if (isLoggedIn()) {
//   }
// };
export { getRequest, deleteRequest, putRequest, postRequest, patchRequest };
