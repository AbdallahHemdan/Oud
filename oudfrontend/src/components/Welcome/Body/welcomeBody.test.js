import React from 'react';
import {shallow, mount} from 'enzyme';
import {findByTestAttr} from './../../../utils/index';
import BeforeLogin from './BeforeLogin';
const setup = (props = {}) => {
  const component = shallow(<BeforeLogin {...props} />);
  return component;
};

describe('Before Login test ', () => {
  let component;
  beforeEach(() => {
    const props = {};
    component = setup(props);
  });
  it('test background in the before login component ', () => {
    const wrapper = findByTestAttr(component, 'background');
    expect(wrapper.length).toBe(1);
  });
  it('test firstText in the before login component ', () => {
    const wrapper = findByTestAttr(component, 'firstText');
    expect(wrapper.length).toBe(2);
  });
  it('test getOudBtn in the before login component ', () => {
    const wrapper = findByTestAttr(component, 'getOudBtn');
    expect(wrapper.length).toBe(1);
  });
  describe('Snapshot test', () => {
    it('renders component correctly', () => {
      const wrapper = shallow(<BeforeLogin />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
