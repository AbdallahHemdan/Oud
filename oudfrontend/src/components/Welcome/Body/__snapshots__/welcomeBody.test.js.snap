// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Before Login test  Snapshot test renders component correctly 1`] = `ShallowWrapper {}`;
